mubench.reviewsite/src/Controllers/RunsController.php

```diff
@@ -603,4 +635,167 @@ private function deleteOldImages($experimentId, $detectorId, $projectId, $versio
             rmdir($path);
         }
     }
+
+    private function createNewDBConnection($host, $database, $username, $password, $prefix)
+    {
+        $this->capsule->addConnection([
+            'driver' => 'mysql',
+            'host' => $host,
+            'database' => $database,
+            'username' => $username,
+            'password' => $password,
+            'charset' => 'utf8',
+            'collation' => 'utf8_unicode_ci',
+            'prefix' => $prefix,
+        ], 'extern');
+        return $this->capsule->getConnection('extern');
+    }
+
+    function importRunsFromConnection($experiment_id, $detector_muid, $project_muid, $version_muid, $connection): void
+    {
+        list($local_detector, $detector) = $this->getLocalAndExternDetector($detector_muid, $connection);
+
+        $run = new Run;
+        $run->setDetector($detector);
+        $run->setConnection($connection->getName());
+
+        if ($project_muid != "" && $version_muid != "") {
+            $runs = $run->in(Experiment::find($experiment_id))->where(['version_muid' => $version_muid, 'project_muid' => $project_muid])->get();
+        } else {
+            $runs = $run->in(Experiment::find($experiment_id))->get();
+        }
+        foreach ($runs as $run) {
+            $cloned_run = $this->importRun($local_detector, $run);
+
+            foreach ($run->misuses as $misuse) {
+                $cloned_misuse = $cloned_run->misuses()->where('misuse_muid', $misuse->misuse_muid)->first();
+                if (!$cloned_misuse) {
+                    list($cloned_misuse, $cloned_metadata) = $this->importMisuse($misuse, $cloned_run, $local_detector);
+                }
+
+                $this->importMisuseFindings($misuse->findings, $local_detector, $cloned_misuse);
+
+                $this->importMisuseSnippets($connection, $misuse);
+
+                if ($misuse->metadata) {
+                    $violationTypes = $this->importMetadataPatternsAndTypes($misuse, $cloned_metadata);
+                    $cloned_metadata->violation_types()->sync($violationTypes);
+                }
+
+                $tag_ids = $this->importTags($misuse);
+                $cloned_misuse->misuse_tags()->sync($tag_ids);
+
+                foreach ($misuse->reviews as $review) {
+                    $this->importReviews($review, $cloned_misuse);
+                }
+            }
+        }
+    }
+
+    private function cloneAndResetConnectionModel(Model $model){
+        $clone = $model->replicate();
```

Comment:
We have three different terms here: 'clone', 'replicate', and 'import'. I'd appreciate having at most two... replace 'clone'/'cloned' by 'import'/'imported'?